pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -118,146 +127,168 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
-	// Create kubevirt datavolume from oVirt VM disks:
-	dvs := newDVForCR(vm, ovirtSecret)
-	/*
-		for _, dv := range dvs {
-			err := r.client.Create(context.TODO(), dv)
-			if err != nil {
-				return reconcile.Result{}, err
-			}
-		}
-	*/
-
-	// Create kubevirt VM from oVirt VM:
+	diskAttachmentsLink, _ := vm.DiskAttachments()
+	diskAttachments, _ := ovirtConnection.FollowLink(diskAttachmentsLink)
+	dvs := newDVForCR(diskAttachments.(*ovirtsdk.DiskAttachmentSlice), ovirtSecret)
 	vmSpec := newVMForCR(vm, dvs, ovirtConnection)
-	err = r.client.Create(context.TODO(), vmSpec)
-	if err != nil {
+
+	// Set VirtualMachineImport instance as the owner and controller
+	if err := controllerutil.SetControllerReference(instance, vmSpec, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
 
-	// Set VirtualMachineImport instance as the owner and controller
-	/*
-		if err := controllerutil.SetControllerReference(instance, pod, r.scheme); err != nil {
-			return reconcile.Result{}, err
-		}
+	// Check if this VM already exists
+	found := &kubevirtv1.VirtualMachine{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: vmSpec.Name, Namespace: vmSpec.Namespace}, found)
+	if err != nil && errors.IsNotFound(err) {
+		reqLogger.Info("Creating a new VM", "VM.Namespace", vmSpec.Namespace, "VM.Name", vmSpec.Name)
 
-		// Check if this Pod already exists
-		found := &corev1.Pod{}
-		err = r.client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
-		if err != nil && errors.IsNotFound(err) {
-			reqLogger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
-			err = r.client.Create(context.TODO(), pod)
+		// Create kubevirt datavolume from oVirt VM disks:
+		for _, dv := range dvs {
+			_, err := r.kubeClient.CdiClient().CdiV1alpha1().DataVolumes("default").Create(&dv) // FIXME:
 			if err != nil {
 				return reconcile.Result{}, err
 			}
+		}
 
-			// Pod created successfully - don't requeue
-			return reconcile.Result{}, nil
-		} else if err != nil {
+		// Create kubevirt VM from oVirt VM:
```

Comment:
Would be great to move vm creation to be first. UI assumes it will be created very fast.